spec/lib/administrate/search_spec.rb

```diff
@@ -59,4 +67,118 @@ class User; end
       end
     end
   end
+
+  describe "#scope" do
+    let(:scope) { "active" }
+    let(:resolver) do
+      double(resource_class: User, dashboard_class: MockDashboard)
+    end
+
+    describe "the query is only the scope" do
+      let(:query) { "#{scope}:" }
+
+      it "returns nil if the model does not respond to the possible scope" do
+        begin
+          class User; end
+          search = Administrate::Search.new(resolver, query)
+          expect(search.scope).to eq(nil)
+        ensure
+          remove_constants :User
+        end
+      end
+
+      it "returns the scope if the model responds to it" do
+        begin
+          class User
+            def self.active; end
+          end
+          search = Administrate::Search.new(resolver, query)
+          expect(search.scope).to eq(scope)
+        ensure
+          remove_constants :User
+        end
+      end
+
+      it "returns nil if the name of the scope looks suspicious" do
+        begin
+          class User
+            def self.destroy_all; end
+          end
+
+          Administrate::Search::BLACKLISTED_WORDS.each do |word|
+            search = Administrate::Search.new(resolver, "#{word}_all:")
+            expect(search.scope).to eq(nil)
+          end
+        ensure
+          remove_constants :User
+        end
+      end
+
+      it "returns nil if the name of the scope ends with an exclamation mark" do
+        begin
+          class User
+            def self.bang!; end
+          end
+
+          search = Administrate::Search.new(resolver, "bang!:")
+          expect(search.scope).to eq(nil)
+        ensure
+          remove_constants :User
+        end
+      end
+
+      describe "with COLLECTION_SCOPES defined" do
+        let(:wresolver) do
+          double(resource_class: User, dashboard_class: DashboardWithDefinedScopes)
+        end
+
+        it "ignores the scope if it isn't included" do
+          begin
+            class User
+              def self.closed; end
+              def self.active; end
+            end
+
+            search = Administrate::Search.new(wresolver, 'closed:')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.